runtime/src/main/kotlin/kotlin/native/internal/GC.kt

```diff
@@ -78,6 +78,19 @@ object GC {
         get() = getTuneThreshold()
         set(value) = setTuneThreshold(value)
 
+    /**
+     * Detect cyclic references going via atomic references and return list of cycle-inducing objects
+     * or `null` if the leak detector is not available.
+     */
+    @SymbolName("Kotlin_native_internal_GC_detectCycles")
+    external fun detectCycles(): Array<Any>?
+
+    /**
+     * Find a cycle starting from the given object, `null` if there are no cycles.
```

Comment:
"Cycle" is somewhat ambiguous. Maybe add "reference"?
Also not sure if a cycle has any start.